src/workerd/api/web-socket.c++

```diff
@@ -790,6 +838,7 @@ kj::Promise<void> WebSocket::pump(
       }
       KJ_CASE_ONEOF(close, kj::WebSocket::Close) {
         co_await ws.close(close.code, close.reason);
+        autoResponse.isClosed = true;
```

Comment:
I think we should be referring to `native.closedOutgoing` instead, no? Feels like we can get rid of this property entirely.

Reply:
Oh I see, is the issue that you might not be able to access `native` when calling `sendAutoResponse()`? In that case, you may also want to set `autoResponse.isClosed = native.closedOutgoing` up in `initNative()`, otherwise we run the risk of a situation where:
- DO sends close, `autoResponse.isClosed = true`, client doesn't ack.
- DO is evicted so `api::WebSocket` GC'd
- `getWebSockets()` is called bringing the `api::WebSocket` back
- client sends auto-request and `autoResponse.isClosed` is `false` by default.